auth/src/user.rs

```diff
@@ -94,20 +175,78 @@ pub struct User {
     pub name: AccountName,
     pub key: ApiKey,
     pub account_tier: AccountTier,
+    pub subscription_id: Option<SubscriptionId>,
 }
 
 impl User {
     pub fn is_admin(&self) -> bool {
         self.account_tier == AccountTier::Admin
     }
 
-    pub fn new(name: AccountName, key: ApiKey, account_tier: AccountTier) -> Self {
+    pub fn new(
+        name: AccountName,
+        key: ApiKey,
+        account_tier: AccountTier,
+        session_id: Option<SubscriptionId>,
+    ) -> Self {
         Self {
             name,
             key,
             account_tier,
+            subscription_id: session_id,
         }
     }
+
+    /// In case of an existing subscription, check if valid.
+    async fn subscription_is_valid(&self, client: &stripe::Client) -> Result<bool, Error> {
+        if let Some(subscription_id) = self.subscription_id.as_ref() {
+            let subscription = stripe::Subscription::retrieve(client, subscription_id, &[]).await?;
+            debug!("subscription: {:#?}", subscription);
+            return Ok(subscription.status == SubscriptionStatus::Active
+                || subscription.status == SubscriptionStatus::Trialing);
+        }
+
+        Ok(false)
+    }
+
+    // Synchronize the tiers with the subscription validity.
+    async fn sync_tier(&mut self, user_manager: &UserManager) -> Result<bool, Error> {
+        let subscription_is_valid = self
+            .subscription_is_valid(&user_manager.stripe_client)
+            .await?;
+        debug!(%subscription_is_valid, "to be or not to be");
```

Comment:
To keep or not to keep this message, that is the question :smile:

Reply:
Removed it 😆